### ISSUE TITLE
[BUG] Conda install fails due to possibly wrong Arrow requirements?

### ISSUE THREAD

**Issue by buhrmann** [rapidsai/cudf#8769]
When trying to install into an existing conda environment, with a recent version of Arrow, the install fails complaining it requires  `pyarrow >=1.0.1,<1.0.2.0a0` which is a very old version. The same happens when using conda itself or mamba. E.g. using mamba, which usually gives more useful feedback than conda:

```
mamba install -c rapidsai -c nvidia -c conda-forge cudf=21.06 cudatoolkit=11.0

Looking for: ['cudf=21.06', 'cudatoolkit=11.0']

conda-forge/linux-64     [====================] (00m:00s) No change
rapidsai/noarch          [====================] (00m:00s) No change
rapidsai/linux-64        [====================] (00m:00s) No change
nvidia/linux-64          [====================] (00m:00s) No change
nvidia/noarch            [====================] (00m:00s) No change
conda-forge/noarch       [====================] (00m:00s) No change
pkgs/r/noarch            [====================] (00m:00s) No change
pkgs/main/linux-64       [====================] (00m:00s) No change
pkgs/r/linux-64          [====================] (00m:00s) No change
pkgs/main/noarch         [====================] (00m:00s) No change

Pinned packages:
  - python 3.8.8
  - blas * openblas
  - catboost 0.24.4.*
  - emoji >=0.4.5,<1.0.0
  - gensim 3.8.3.*
  - joblib 0.17.0.*
  - numpy 1.19.5.*
  - pandas 1.2.3.*
  - pip 20.2.*
  - pyarrow 4.0.1.*
  - pynndescent 0.5.2.*
  - python 3.8.8.*
  - python-annoy 1.16.3.*
  - python-dateutil <2.8.1
  - scikit-learn 0.24.1.*
  - scipy 1.6.0.*
  - spacy 3.0.1.*
  - umap-learn 0.5.1.*
  - xlrd 1.2.0.*


Encountered problems while solving:
  - package cudf-21.06.00-cuda_11.2_py38_gae440465c2_0 requires pyarrow >=1.0.1,<1.0.2.0a0, but none of the providers can be installed
```

Searching for information about the dependencies of `cudf`, this seems to be a valid complaint by the solver:

```
mamba info cudf=21.06.00=cuda_11.2_py38_gae440465c2_0

...

cudf 21.06.00 cuda_11.2_py38_gae440465c2_0
------------------------------------------
file name   : cudf-21.06.00-cuda_11.2_py38_gae440465c2_0.tar.bz2
name        : cudf
version     : 21.06.00
build string: cuda_11.2_py38_gae440465c2_0
build number: 0
channel     : https://conda.anaconda.org/rapidsai/linux-64
size        : 109.9 MB
arch        : x86_64
constrains  : ()
license     : Apache-2.0
license_family: Apache
md5         : 888b09b73a9df4d65fb7613a8998b138
platform    : linux
subdir      : linux-64
timestamp   : 1623271240405
url         : https://conda.anaconda.org/rapidsai/linux-64/cudf-21.06.00-cuda_11.2_py38_gae440465c2_0.tar.bz2
dependencies:
    cachetools
    cudatoolkit >=11.2.72,<11.3.0a0
    cupy >7.1.0,<10.0.0a0
    fastavro >=0.22.0
    fsspec >=0.6.0
    libcudf >=21.6.0,<21.7.0a0
    numba >=0.53.1
    numpy
    nvtx >=0.2.1
    packaging
    pandas >=1.0,<1.3.0dev0
    protobuf
    pyarrow >=1.0.1,<1.0.2.0a0
    python >=3.8,<3.9.0a0
    python_abi 3.8.* *_cp38
    rmm >=21.6.0,<21.7.0a0
    typing_extensions
```

**Steps/Code to reproduce bug**
On Linux, create an environment with pyarrow pinned to some "recent" version and try to install cudf with conda/mamba. 

**Expected behavior**
Installs in a way compatible with recent versions of pyarrow (e.g. >=2.x). I don't imagine there is a particular reason to need that old a version of Arrow?

**Environment overview (please complete the following information)**
 Attempt to install with `mamba install -c rapidsai -c nvidia -c conda-forge cudf=21.06 cudatoolkit=11.0` in a debian-based docker container.

**Comment by beckernick** [comment]
Hi @buhrmann , thanks for filing an issue. The Arrow requirements are a known limitation in 21.06 that has been resolved in the 21.08 nightlies and will be available in the upcoming 21.08 release.

If you'd like to use the nightlies, you can follow our [release selector](https://rapids.ai/start#get-rapids) to configure for your system.

I'm going to close this issue as answered, but please feel free to re-open if you run into further issues.

**Comment by buhrmann** [comment]
Great! Thanks for the quick response. Can confirm that installing from nightly worked without problem 👍🏼
